pennylane/decomposition/decomposition_graph.py

```diff
@@ -208,88 +215,65 @@ def _add_decomp_rule_to_op(
         except DecompositionNotApplicable:
             pass  # ignore decompositions that are not applicable to the given op params.
 
-    def _add_adjoint_decomp_node(self, op_node: CompressedResourceOp, op_node_idx: int) -> int:
-        """Adds an adjoint decomposition node."""
+    def _get_adjoint_decompositions(self, op: CompressedResourceOp) -> list[DecompositionRule]:
```

Comment:
What about using something like `resource_op` when we are dealing with a `CompressedResourceOp`. Might make it a little clearer what of object we are dealing with, as I tend to associate `op` with `Operator`.

Reply:
Sure